### ISSUE TITLE
[Bug]: Tried to log in "user" but could not verify token

### ISSUE THREAD

**Issue by smart7324** [nextcloud/server#37492]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] Nextcloud Server **is** running on 64bit capable CPU, PHP and OS.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

As soon as I open Nextcloud in a new tab, I get redirected to login page and have to login again. Then always the first login fails/nothing happens, so I have to login twice. I am seeing lots of "Tried to log in "user" but could not verify token" errors in log.

It is only happening on Safari (macOS, iOS, iPadOS), tried several versions, also did a clean install of Nextcloud 26 and still the same. Also tried with another user account on a different Mac.

At first I thought it could be related to #33919, but it doesn't seem to be the case. I really spent many hours in trying to get this fixed, but I have no clue, why it is not working.

### Steps to reproduce

1. Login to Nextcloud in Safari
2. Open another tab and open Nextcloud (alternatively close browser and open it again)
3. You will be redirected to login page and the message "Tried to log in "user" but could not verify token" is in log file.

### Expected behavior

The user should still be logged in and not be redirected to login page.

### Installation method

Community Manual installation with Archive

### Nextcloud Server version

26

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.1

### Web server

Apache (supported)

### Database engine version

MySQL

### Is this bug present after an update or on a fresh install?

Fresh Nextcloud Server install

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "***REMOVED SENSITIVE VALUE***"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "26.0.0.11",
        "overwrite.cli.url": "***REMOVED SENSITIVE VALUE***",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "maintenance": false,
        "mail_smtpsecure": "tls",
        "mail_sendmailmode": "smtp",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "587",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "csrf.disabled": true,
        "integrity.check.disabled": true,
        "logfile": "\/var\/www\/cloud\/data\/nextcloud.log",
        "loglevel": 4,
        "enable_previews": true,
        "remember_login_cookie_lifetime": 31536000,
        "session_lifetime": 31536000,
        "session_relaxed_expiry": true,
        "session_keepalive": true,
        "simpleSignUpLink.shown": false,
        "htaccess.IgnoreFrontController": true,
        "default_phone_region": "DE",
        "default_language": "de",
        "force_language": "de",
        "theme": "***REMOVED SENSITIVE VALUE***",
        "defaultapp": "files",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "updater.release.channel": "stable"
    }
}
```

### List of activated Apps

```shell
Enabled:
  - cloud_federation_api: 1.9.0
  - comments: 1.16.0
  - dav: 1.25.0
  - federatedfilesharing: 1.16.0
  - files: 1.21.1
  - files_pdfviewer: 2.7.0
  - files_rightclick: 1.5.0
  - files_sharing: 1.18.0
  - files_versions: 1.19.1
  - logreader: 2.11.0
  - lookup_server_connector: 1.14.0
  - notes: 4.7.2
  - notifications: 2.14.0
  - oauth2: 1.14.0
  - password_policy: 1.16.0
  - provisioning_api: 1.16.0
  - related_resources: 1.1.0-alpha1
  - settings: 1.8.0
  - sharebymail: 1.16.0
  - systemtags: 1.16.0
  - theming: 2.1.1
  - theming_customcss: 1.13.0
  - twofactor_backupcodes: 1.15.0
  - updatenotification: 1.16.0
  - viewer: 1.10.0
  - workflowengine: 2.8.0
Disabled:
  - activity: 2.18.0 (installed 2.14.3)
  - admin_audit: 1.16.0
  - bruteforcesettings: 2.6.0 (installed 2.3.0)
  - circles: 26.0.0 (installed 26.0.0)
  - contactsinteraction: 1.7.0 (installed 1.7.0)
  - dashboard: 7.6.0 (installed 7.1.0)
  - encryption: 2.14.0
  - extract: 1.3.5 (installed 1.3.5)
  - federation: 1.16.0 (installed 1.16.0)
  - files_external: 1.18.0
  - files_texteditor: 2.15.0 (installed 2.15.0)
  - files_trashbin: 1.16.0 (installed 1.11.0)
  - firstrunwizard: 2.15.0 (installed 2.15.0)
  - nextcloud_announcements: 1.15.0 (installed 1.15.0)
  - photos: 2.2.0 (installed 1.3.0)
  - privacy: 1.10.0 (installed 1.10.0)
  - recommendations: 1.5.0 (installed 1.0.0)
  - serverinfo: 1.16.0 (installed 1.12.0)
  - support: 1.9.0 (installed 1.9.0)
  - survey_client: 1.14.0 (installed 1.9.0)
  - suspicious_login: 4.4.0
  - text: 3.7.2 (installed 3.3.0)
  - twofactor_totp: 8.0.0-alpha.0
  - user_ldap: 1.16.0
  - user_status: 1.6.0 (installed 1.1.1)
  - weather_status: 1.6.0 (installed 1.1.0)
```

### Nextcloud Logs

```shell
{"reqId":"***REMOVED SENSITIVE VALUE***","level":1,"time":"2023-03-30T11:50:23+00:00","remoteAddr":"***REMOVED SENSITIVE VALUE***","user":"--","app":"core","method":"GET","url":"/login","message":"Tried to log in user but could not verify token","userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.4 Safari/605.1.15","version":"26.0.0.11","data":{"app":"core"}}
```

**Comment by JohnEridium** [comment]
I also had this issue today and I could only fix it with a database maintenance run (command below). 
my environment infos:

`root@Nextcloud:~# apache2 -v       
Server version: Apache/2.4.41 (Ubuntu)
Server built:   2023-03-08T17:32:54
root@Nextcloud:~# php --version
PHP 8.1.17 (cli) (built: Mar 16 2023 14:38:17) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.1.17, Copyright (c) Zend Technologies
    with Zend OPcache v8.1.17, Copyright (c), by Zend Technologies
root@Nextcloud:~# lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.6 LTS
Release:        20.04
Codename:       focal
root@Nextcloud:~# cat /var/www/nextcloud/version.php 
<?php 
$OC_Version = array(26,0,0,11);
$OC_VersionString = '26.0.0';
$OC_Edition = '';
$OC_Channel = 'stable';
$OC_VersionCanBeUpgradedFrom = array (
  'nextcloud' => 
  array (
    '25.0' => true,
    '26.0' => true,
  ),
  'owncloud' => 
  array (
    '10.11' => true,
  ),
);
$OC_Build = '2023-03-21T09:23:03+00:00 62cfd3b4c9ff4d8cdbbe6dcc8b63a1085bb94e3d';
$vendor = 'nextcloud';`

(PostgreSQL) 12.14 (Ubuntu 12.14-0ubuntu0.20.04.1)


How I fix the loop:
`alias FIX_LOOP='cd /var/www/nextcloud && sudo -u www-data php ./occ maintenance:repair'`
and then wait 30 minutes for the rate limiting to cool down. 

iOS is the latest 16.04 (20E247)

here is an excerpt from my logs when I tried to log in with my admin account:
<img width="359" alt="Screenshot 2023-04-04 at 21 22 02" src="https://user-images.githubusercontent.com/40388847/229899500-70f68f22-c648-454f-ab44-1049aa6f0913.png">


Please answer to this if I should provide more info

**Comment by Hubsi3** [comment]
I gave it a try, but this didn't work for me. Same issue. It also happened to me on a clean new install. So we definitely need help here. At this time NC is completely unusable on Safari no matter what apple device...

**Comment by Yetangitu** [comment]
(moved from #33919)

This problem does not seem to have been solved in v26.0.0.11 - even though https://github.com/nextcloud/server/pull/35419 was merged - seeing how as I'm currently unable to login using Firefox/Android on a device which had a single tab open yesterday. Deleting site data does not change this, nor does running occ maintenance:repair.

I _can_ login using a different browser but not with Firefox, all I get is an empty page showing the site logo and the footer - there is no error message but no login/password request either.

This does _not_ work:

 - deleting all site data and cookies for the domain
 - force-stopping and restarting Firefox/Android
 - rebooting the device
 - trying different network connections - wifi, 4G, VPN
 - deleting browsing history for the affected domain
 - using another open session to delete all sessions for the affected device (in _Settings->Security->Devices & Sessions_)
 - updating Firefox/Android (to 111.0)
 - running `occ maintenance:repair`
 - burning black candles in a fairy ring in the forest while chanting obscure incantations (well, did not try but I don't think it would work)

This _does_ work:

 - using a different browser
 - using private mode in Firefox

The error message in the log is the one which has been shown countless times already: `Tried to log in "username" but could not verify token`:

```
{"reqId":"aupvuif3Msicz86FxhbY","level":1,"time":"April 06, 2023 06:06:04","remoteAddr":"192.168.9.2","user":"--","app":"core","method":"GET","url":"/login","message":"Tried to log in frank but could not verify token","userAgent":"Mozilla/5.0 (Android 9; Mobile; rv:109.0) Gecko/112.0 Firefox/112.0","version":"26.0.0.11","data":{"app":"core"}}
{"reqId":"q8JEudtB0oT3gfNqLYye","level":1,"time":"April 06, 2023 06:06:04","remoteAddr":"192.168.9.2","user":"--","app":"core","method":"GET","url":"/apps/theming/image/background?v=27","message":"Tried to log in frank but could not verify token","userAgent":"Mozilla/5.0 (Android 9; Mobile; rv:109.0) Gecko/112.0 Firefox/112.0","version":"26.0.0.11","data":{"app":"core"}}
```
The _really_ annoying thing is that I do not get a chance to login at all since the login/password request does not show up - only the site logo and the footer on an otherwise empty page.

Move: https://github.com/nextcloud/server/issues/37492

**Comment by Yetangitu** [comment]
Another thing which _does_ work:

1.  enable USB debugging in Firefox/Android
2. connect it to another machine though USB
3. open the debugger on the Nextcloud tab
4. got to the _Network_ section
5. make sure that 'Disable cache' is checked
6. reload the tab

This way I _do_ get a login/password request. It seems that Firefox' _Clear cookies and site data_ is not enough to actually clear everything related to the page.

**Comment by JohnEridium** [comment]
Update: This now happens multiple times per day, which is a lot worse than it was before updating to NC 26

**Comment by Hubsi3** [comment]
This is really a serious issue. Right now, I can't use NC with Safari... I am getting logged out every page refresh, so it's completely unusable. Are there any updates? :)

**Comment by mafjensengithub** [comment]
I have some home-made apps, and also have issues when using safari. Most from time to time (rather often) the Axios get calls fails.

Now I just added a timestamp to all my Axios get's, and that seems to solve a lot of the issues regarding Safari. Could it be some internal caching on the browser side for axios calls, where Safari might behave differently than other browsers (Who would expect that :-))

**Comment by mafjensengithub** [comment]
This seems to be originating in the way tokens are renewed in session.php. I have some apps written with 4 axios get calls and from Safari these are fired in paralell using the same session, and hance this the same token. In session.php line 889, session tokens are replaced after first use, and hence this, all the next axios gets are failing as they still all contains the original token from when the first was fired.

The way of replacing tokens after 1 use seems like a complex scheme on a site utilizing multiple async calls more or less in paralell.

**Comment by mafjensengithub** [comment]
Some of the new issues could be related to a safari bug: https://bugs.webkit.org/show_bug.cgi?id=255524

**Comment by JohnEridium** [comment]
Maybe iOS 17 brings a change or the root cause is found somewhere else, either way I hope this will soon be solved because sometimes I can’t log into my NC for days

**Comment by smart7324** [comment]
Seems to be fixed for me with iOS 16.5 and macOS 13.4.

**Comment by JohnEridium** [comment]
Updated a few days ago and for me it seems to be just as bad as before. Haven’t replied earlier because I wanted to gather some data.

**Comment by smart7324** [comment]
I'm not experiencing any issues, also on NC 27.0.0. We can close here.

**Comment by JohnEridium** [comment]
I updated ~12h ago and just had this issue reappear. Setup is NC in a Ubuntu 11 LXC run on Proxmox 7.4-3.

Kernel: 5.15.107-2-pve
Ubuntu: Ubuntu 20.04.6 LTS
PHP: PHP 8.1.17 (cli) (built: Mar 16 2023 14:38:17) (NTS) Copyright (c) The PHP Group Zend Engine v4.1.17, Copyright (c) Zend Technologies with Zend OPcache v8.1.17, Copyright (c), by Zend Technologies
Apache Server version: Apache/2.4.41 (Ubuntu)
NC version: 27.0.0.8

Reverse Proxy: Nginx-Proxy-Manager
RP version: 2.10.3

Client: iOS 16.5 - Safari
<img width="798" alt="Screenshot 2023-06-16 at 07 51 26" src="https://github.com/nextcloud/server/assets/40388847/52e24b3f-fa3f-4771-90c1-963e9aa7f75b">
<img width="801" alt="Screenshot 2023-06-16 at 07 52 19" src="https://github.com/nextcloud/server/assets/40388847/0fc3d229-2cae-49c4-8729-f6be67935f6c">
<img width="343" alt="Screenshot 2023-06-16 at 07 50 11" src="https://github.com/nextcloud/server/assets/40388847/cdf64209-2797-4887-85cc-74ea8c92c9a6">



Did you do anything else than simply updating NC to fix this? It is getting more and more frustrating to use NC since I can't access it ~50% of the time I need to

**Comment by smart7324** [comment]
Okay hm, I also didn’t experience the bug on NC 26 since iOS 16.5… I did not change anything, but it’s just working.

So I reopen this issue for you.

**Comment by JohnEridium** [comment]
Thanks a lot 

that’s weird… Do you think my or any reverse proxy could be an issue since my TLS connection is terminated there?
I can’t really think of anything else that could cause this in my setup

**Comment by smart7324** [comment]
Honestly I don't think so, as I also had this issue and don't have a reverse proxy. I also did some debugging, but I haven't found anything...
Is it working with other browsers for you?

**Comment by JohnEridium** [comment]
I rarely use other devices to access my NC, but I had a few situations where this error occurred with my employer provided laptop. 
On my Debian laptop with Firefox, I had a kinda similar error where i was kinda logged in, but was repeatedly kicked out of NC with the error message in the browser „you are not logged in“. Even when I logged out and back in, this error would persist. I blamed a weird cookie issue and just let it be.

Another possibility that just came to mind: I’m basically always connected to my VPN server at home, which gives my phone, my Mac and NC the same public IP address. Could this be an issue?

**Comment by smart7324** [comment]
Hm very interesting… Sorry, but I don’t know if your ip can be a source of the issue. Maybe someone else can help?

**Comment by JohnEridium** [comment]
Yeah me neither, I’m just throwing guesses at the wall here to see what sticks haha

To anyone reading this: all suggestions are welcome 


Btw, I played around on my work phone (also iPhone and safari) and was able to provoke the error relatively quickly with two open tabs and some reloads/NC-App switching 
The error occurred but I was not logged out however, that also happens a lot

**Comment by MrRies** [comment]
Hi,
We have also been struggling with this problem for about two months. Even an update to version 27 has not brought any improvement. On the contrary, we have the feeling that the bug has increased significantly in recent weeks.
In the meantime, our power users can no longer use Nextcloud on certain days.

Even deleting the cookies only helps to a limited extent. After deleting them, they are simply set again and the problem is back.

Our Nextcloud is connected to a very large LDAP directory of our institution. We have about 70 active users (once a week) and about 20 power users (every day, several hours). We are thinking that a connection to the LDAP could be increasing the problem, but probably the trigger is somewhere else.

Access is via a reverse proxy (nginx). There, too, we have already changed some settings for header modification, but without any noticeable effect.
In addition, the token errors are occurring more and more frequently with reports of a brute force attack. For this reason, we have to deactivate the brute force detection in the meantime in order not to be locked out all the time. Apparently, Nextcloud counts every expired cookie as a failed login.

It is frustrating. The error pattern is so varied that it is difficult for us to identify the origin of the error.

**Comment by JohnEridium** [comment]
Yes, that’s also my experience
And that’s on a very small instance with only me as a user. 
what client devices do your users use? Maybe we have an overlap and can help narrow down the scope for the devs

**Comment by JohnEridium** [comment]
yeah, that's pretty much my device variety, just at a way smaller scale. I have an iPhone with iOS 16.5.1(c), an iMac with macOS 13.2.1 (both most recent Safari), Laptop with Debian 11 + Firefox and a Windows laptop from my employer with Windows 10, now windows 11. All devices had the issue with always the same symptoms that have been described here multiple times. 

Just now I updated my NC instance to 27.0.1 and I could not immediately provoke the error. I'll report back in a few days if the issue surfaces again.

**Comment by JohnEridium** [comment]
update after a few days: 

the message "tried to log in $USER but could not verify token" still appears when tabs get reactivated (browser opened after some Time on iOS e.g.) or occasionally when you have multiple tabs open, but no issues as in kicked out, rate limited and unable to log in.

So from my user+admin perspective, the symptoms are mended but the cause still persists in some form. Since it's on multiple OSs _and_ different browsers on these _and_ NC changed behaviour after an update, I assume the issue is still somewhere in NC. 

I'll post another update in like 2 weeks or when I am facing the initial issues again.

anything new in your setup? @MrRies

**Comment by JohnEridium** [comment]
Update from my setup:
It's happening rarely, but still relatively predictable. For example, a tab has been open for a few days but not active for most of this time, then another gets opened. This, in my case, triggers the spew of "tried to log in $USER but could not verify token" messages (see screenshot).

Even after the brute force cooldown and a successful login, I get these log messages but without a kick.

Would be great if anyone from the team took a look in here to tell us what other infos to provide. My setup did not change from [this](https://github.com/nextcloud/server/issues/37492#issuecomment-1646331060) message.

**Comment by JohnEridium** [comment]
Never mind, my loose theory was proven wrong unsurprisingly. As soon as I have one „older“ tab and open another, I get kicked out immediately. 

It seems NC is messing up the cookies with different tabs, like it tries to verify tab1 with cookie 2 and thus throws errors.

I just don’t know how to troubleshoot this or what information to provide from where

**Comment by MrRies** [comment]
Sorry for the long absence.
We also updated to 27.0.1, but this did not lead to any change.

Our access is via a subdomain using a reverse proxy (nginx). Out of sheer desperation, we tried deactivating the settings "Block Common Exploits" and "Cache assets" in the nginx proxy manager for this subdomain.
This resulted in the error message appearing just as often, but it no longer has any consequences for the users.
We can currently work in several tabs again without any problems.
However, the problem is so random that we are not sure whether this was really the cause of the problem or pure coincidence.

**Comment by JohnEridium** [comment]
> Sorry for the long absence. We also updated to 27.0.1, but this did not lead to any change.
> 
> Our access is via a subdomain using a reverse proxy (nginx). Out of sheer desperation, we tried deactivating the settings "Block Common Exploits" and "Cache assets" in the nginx proxy manager for this subdomain. This resulted in the error message appearing just as often, but it no longer has any consequences for the users. We can currently work in several tabs again without any problems. However, the problem is so random that we are not sure whether this was really the cause of the problem or pure coincidence.

that's a valuable tip, I'll try to deactivate that as well

**Comment by dafi87** [comment]
We are seeing the same problem with NC 27.0.1 on PHP 8.1

Log gets flooded every some seconds with "Tried to log in $USER but could not verify token". Only very few users are affected by it.

**Comment by JohnEridium** [comment]
I applied the "trick" @MrRies suggested (deactivate caching, web sockets and 'block common exploits' in Nginx proxy manager) and it seems to remedy the symptoms i.e. being kicked out. However I still have the messages in the log, but for some reason NC doesn't lock me out anymore.

So, it could be a (hopefully) temporary fix until the actual root issue is fixed.

**Comment by ChristophWurst** [comment]
Discovered https://github.com/spring-projects/spring-security/issues/11921 and it could be a similar issue. Nextcloud uses a token-based remember me mechanism. Two or more requests may race for the same database token.

**Comment by ChristophWurst** [comment]
Attempt on a simpler, hopefully more robust session handling: https://github.com/nextcloud/server/pull/40543

**Comment by ChristophWurst** [comment]
If someone is okay with experimenting on a production system see https://github.com/nextcloud/server/pull/40628.

Nextcloud 27.x patch can be fetched from https://github.com/nextcloud/server/commit/37eefca9f11eb4fe0ba6a662ad642565e6843bac.patch.

This will harden the token refresh logic and log more specific errors when refreshing tokens doesn't work.

This should give us valuable insights into the failing remember-me login.

**Comment by JohnEridium** [comment]
I‘d be up to test this but I’m not sure how to apply this patch… 
Is there a guide in the documentation somewhere on how to do this?

**Comment by ChristophWurst** [comment]
https://docs.nextcloud.com/server/latest/admin_manual/issues/applying_patch.html

**Comment by JohnEridium** [comment]
And how do I get the "patch" command? 
Sorry, I've never seen that command before and looking for "ubuntu install patch GitHub" doesn't exactly yield the results I'm looking for haha

**Comment by ChristophWurst** [comment]
No worries. Let's wait for a tech-savvy person to test the patch and report findings based on logfile analysis.

**Comment by JohnEridium** [comment]
ah nvm, I thought patch was a php script like occ. I applied the patch and I'm changing the log level to debug

will report back when I find anything

**Comment by JohnEridium** [comment]
<img width="663" alt="Screenshot 2023-09-27 at 10 10 09" src="https://github.com/nextcloud/server/assets/40388847/ff0795b8-4f10-40e8-98c0-b959a8a871b7">

![Screenshot 2023-09-27 at 10 07 27](https://github.com/nextcloud/server/assets/40388847/f687b7f7-855c-4aad-8d4f-afe2a92927d2)

The only thing I could see immediately; I think the screenshot with the two tokens was a remnant from before the patch apply because I logged in from my work phone.

About the "No public access[...]" messages: do you have an idea what could cause this? DAV sounds like calendar or contacts.

exception trace:

`{file=/var/www/nextcloud/3rdparty/sabre/event/lib/WildcardEmitterTrait.php, line=89, function=beforeMethod, class=Sabre\DAV\Auth\Plugin, type=->}, {file=/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php, line=456, function=emit, class=Sabre\DAV\Server, type=->}, {file=/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php, line=253, function=invokeMethod, class=Sabre\DAV\Server, type=->}, {file=/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php, line=321, function=start, class=Sabre\DAV\Server, type=->}, {file=/var/www/nextcloud/apps/dav/lib/Server.php, line=365, function=exec, class=Sabre\DAV\Server, type=->}, {file=/var/www/nextcloud/apps/dav/appinfo/v2/remote.php, line=35, function=exec, class=OCA\DAV\Server, type=->}, {file=/var/www/nextcloud/remote.php, line=172, args=[/var/www/nextcloud/apps/dav/appinfo/v2/remote.php], function=require_once}`

**Comment by JohnEridium** [comment]
Activated an already open tab after some time, this is what happens:

<img width="684" alt="Screenshot 2023-09-27 at 11 13 41" src="https://github.com/nextcloud/server/assets/40388847/868920cc-0743-4eab-a087-8967326c352b">

**Comment by ChristophWurst** [comment]
I think there are some requests that fail, other succeed. It's a bit random. Sometimes that ends your session, other times it can survive.

Thanks for the insights.

**Comment by JohnEridium** [comment]
Sure, any time
just ping me when you have anything new, I'm happy to test stuff

**Comment by JohnEridium** [comment]
Addendum: I just opened a new NC tab on my phone (closed the ones from earlier after testing the patch) and had to log in again

 it’s not new behavior, but I had to enter my username and PW twice and then the 2FA code before it let me in

idk when exactly I was kicked out but when I find the section in the logs I’ll post a screenshot

**Comment by ChristophWurst** [comment]
Double login is a known issue and related to lost sessions

**Comment by ChristophWurst** [comment]
Could you please apply https://github.com/nextcloud/server/commit/02591953bc488aa424f058035ad39fa7b3beb723.patch as well? It's an amendment to https://github.com/nextcloud/server/pull/40628 so that it logs the request that wins the race for the token.

**Comment by JohnEridium** [comment]
Will do when I’m home

**Comment by JohnEridium** [comment]
[All-Messages-search-result-part-1.csv](https://github.com/nextcloud/server/files/12743251/All-Messages-search-result-part-1.csv)
[All-Messages-search-result-part-2.csv](https://github.com/nextcloud/server/files/12743252/All-Messages-search-result-part-2.csv)


I applied the patch and played around a bit (I redacted any tokens, IPs and domains).
Is this format helpful for you? Should I test specific scenarios or filter for certain keywords? Because right now this is basically the full log after the patch apply and reboot on debug level.

**Comment by ChristophWurst** [comment]
The format is fine. Thanks!

First:
```
2023-09-27T22:25:56.086+02:00;Nextcloud;Remember-me token TOKEN1/some_extras?/ for root replaced by TOKEN2
```
Later:
```
2023-09-27T22:28:02.133+02:00;Nextcloud;Tried to log in root but could not find token TOKEN3 in database
2023-09-27T22:28:02.133+02:00;Nextcloud;Tried to log in root but could not find token TOKEN3 in database
2023-09-27T22:28:02.133+02:00;Nextcloud;Tried to log in root but could not find token TOKEN3 in database
2023-09-27T22:28:02.133+02:00;Nextcloud;Tried to log in root but could not find token TOKEN3 in database
```
But then also
```
2023-09-27T22:28:02.133+02:00;Nextcloud;Remember-me token TOKEN3 for root replaced by TOKEN4
```

so token2 is never used. token3 appears out of nowhere.

Did you have more than one browser or devices connected? e.g. desktop+phone.

**Comment by JohnEridium** [comment]
Ah yes, my bad

I was just focused on provoking the error and didn't think about multiple devices. I just recreated the situation with just one device but the overall situation seems to be the same:

<img width="755" alt="Screenshot 2023-09-28 at 09 00 47" src="https://github.com/nextcloud/server/assets/40388847/7e66d476-fd3c-42ee-995c-2c1f30160245">

What I did:
closed the NC app on my Mac and closed tabs on my work phone. Then after some minutes I opened Safari on my private phone, closed an inactive tab and opened a new one. Only when I opened the new tab, logs started appearing. 
Once the new tab loaded, I logged in (twice, as noted yesterday) and you see the logs above.

**Comment by JohnEridium** [comment]
a little piece of info that may or may not be relevant:

a few days ago, a friend wanted to make an account on my NC instance and after a relatively log waiting time on the signup page, got an error. in the logs it said something about that the signup mail could not be sent. Curiously though, an email was actually sent and my friend went through the signup process successfully despite the errors.

I don't know the NC stack and processes well enough to make a qualified guess, but it feels a bit like some more or less central loop or routine is run multiple times, that writes to the database. Maybe this gives you a clue where to look...

<img width="792" alt="Screenshot 2023-10-08 at 16 18 14" src="https://github.com/nextcloud/server/assets/40388847/fd26e4bd-60cf-40f8-a23b-532b8862a9b4">

**Comment by ChristophWurst** [comment]
password reset is off-topic. look for existing tickets or create a new one if there is none. Attached the log entries as text please to make it searchable.

**Comment by JohnEridium** [comment]
Thought this may be a related issue or a symptom of the same root cause
Sorry for the extra message, please ignore

**Comment by KenBW2** [comment]
I am experiencing frequent logouts, but I don't believe it's the same issue reported here, since I don't get the "Tried to login but couldn't verify token" error

In fact I don't get an error at all. I did open a new ticket https://github.com/nextcloud/all-in-one/issues/3674 but that's been closed and directed here.

I've noticed that sometimes I get this issue where the header is logged in, and the content is logged out:

![Screenshot from 2023-11-16 00-24-30](https://github.com/nextcloud/server/assets/366540/baf40138-f2f9-47a6-86de-8dfdd71165cb)

**Comment by KenBW2** [comment]
Today I was logged out and saw the login page without the header being logged in.

Opening a new tab didn't have me logged in again

**Comment by ChristophWurst** [comment]
We can't pinpoint it but https://github.com/nextcloud/server/pull/40879, https://github.com/nextcloud/server/pull/41318 or a related change must have helped with this problem because it now happens seldom. The theory is that we use `session_regenerate_id` less often, which is documented to be problematic at https://www.php.net/manual/en/function.session-regenerate-id.php

> Warning: Currently, session_regenerate_id does not handle an unstable network well, e.g. Mobile and WiFi network. Therefore, you may experience a lost session by calling session_regenerate_id.

**Comment by cdauth** [comment]
Wanted to mention that I'm seeing exactly the symptoms described here, but in Chromium and only after waking up from standby. I always have a few Nextcloud tabs open in my browser, and I assume at least some of them are polling the server and keep me logged in. When my computer goes to standby and wakes up again, the different Nextcloud apps that are open show error notifications or misbehave in different ways, and when I reload the page I get forwarded to the login page. The log shows a `Tried to log in but could not verify token` error for every resource that the browser tried to load.

On the login page I have to log in twice. After the first attempt, the login page simply shows up again without any error. The second attempt works properly. It does not make a difference whether I type the right password on the first attempt. After the first attempt, I see `IP address throttled because it reached the attempts limit in the last 30 minutes [action: login, delay: 200, ip: ...]` on the log and `Bruteforce attempt from \"...\" detected for action \"login\".`

**Comment by JohnEridium** [comment]
What NC version are you on which and reverse proxy are you using with what settings?
I’m asking because with the latest version, the issue is way less frequent, but not fully gone. 

Also, in the past few days after updating, I’ve had a basic http login pop up sometimes while on mobile when I’m opening a new NC tab. I wanted to investigate further before I comment again here, so if anyone else has this problem, maybe we could crowd source how to reliably reproduce this

**Comment by cdauth** [comment]
I'm running the `stable` docker image, which seem to be version 27.1.4. If there have been any relevant changes recently, I could also upgrade to `latest`.

I'm running the docker image with `APACHE_DISABLE_REWRITE_IP=1` (to let Nextcloud rather than Apache handle the proxy headers). In my Nextcloud config I have
```php
  'trusted_proxies' =>
  array (
    0 => '127.0.0.0/8',
    1 => '172.16.0.0/12',
    2 => '192.168.0.0/16',
    3 => '10.0.0.0/8',
    4 => 'fc00::/7',
  ),
```
Requests are coming in from a traefik reverse proxy running on `10.10.2.1`.

Also, I have `'session_lifetime' => 1209600,` configured, but the issue happens reproducibly every time my laptop was in standby for a few hours.

**Comment by joshtrichards** [comment]
Every one on this thread that thinks they're experiencing this issue:

- Please make sure your proxy is not configured to do any caching (e.g. NPM's *Cache Assets* should be disabled - it's not appropriate for use with Nextcloud Server, if you are using your own proxy configuration please review how you've configured caching in the proxy itself)

This variable (which is an invalid configuration) needs to be eliminated first so that only the logs/symptoms/code paths of any underlying bug(s) are visible.

**Comment by cdauth** [comment]
For my case, I'm pretty sure that traefik doesn't do any caching.

**Comment by GNU-Plus-Windows-User** [comment]
I'm not sure if this is relevant or the best place to post, but I'm posting this here since this is the best thread I can find on the issue. I used to see this exact error in my log on occasion(While also being logged out), but now I do not see it anymore, and my issue has gotten worse (being logged out multiple times a day) ever since upgrading to 28.0.5. I also used to have to log in twice when I was logged out with a "Temporary error please try again" but now I sometimes won't even see that error.
 
I originally thought it may have to do with having Nextcloud Mail(Since I had this happen multiple times with mail open), but it doesn't always happen with Nextcloud Mail so I think it's just a coincidence.

I get errors like these in my logs, but an error isn't always logged when I get logged out.
```
Exception
HMAC does not match.
Could not decrypt or decode encrypted session data
```

I've been also getting these new errors since very recently:
```
InvalidTokenException
Token does not exist: redacted
Renewing session token failed: Token does not exist: redacted
```

This issue doesn't happen with the desktop client or mobile client, only with browsers. I've encountered this issue on both Brave and Graphene OS's Vanadium, but I haven't tested other browsers so I'm not sure if it's all browsers or just specific browsers.

**Comment by yan12125** [comment]
I have been affected the "could not verify token" error for at least months, and such errors seem more common recently. Those errors happen with Firefox on either Arch Linux or Windows.

I tried the latest patch from https://github.com/nextcloud/server/pull/40628 (https://github.com/nextcloud/server/commit/908c381018ba95ee1c13d11d35414db3248782d8.patch) with Nextcloud 28.0.4, and got two errors:

```
Tried to log in yen but ran into concurrent session revival
```

```
OC\\Http\\CookieHelper::setCookie(): Argument #2 ($value) must be of type string, null given, called in /usr/share/webapps/nextcloud/lib/private/User/Session.php on line 1062
```
The second error is from:
https://github.com/nextcloud/server/blob/908c381018ba95ee1c13d11d35414db3248782d8/lib/private/User/Session.php#L1062-L1071
which is understandable as `$token` is `null` due to the first error.

As a record, I use nginx and I believe cache is disabled with the following configuration: (from https://github.com/NginxProxyManager/nginx-proxy-manager/issues/389#issuecomment-895485154)
```nginx
      proxy_no_cache 1;
      proxy_cache_bypass 1;
      proxy_cache off;
```

**Comment by yan12125** [comment]
Got a different error message (redacted)
```
Tried to log in xxx but could not find token yyy in database
```
In the database, the token yyy is indeed missing. I got no results with:
```SQL
select * from nextcloud.oc_preferences where appid = 'login_token' and configKey = 'yyy';
```

**Comment by yan12125** [comment]
> I have a new idea. What if the remember-me logic does its job but the concurrent requests cause the web session to be deleted from the database? that would also end a session. I'll prepare some more logging patches 😩

Concurrent requests may not be the (only) issue. I still get the following error after disabling HTTP/2 altogether. I assume there are no concurrent requests with HTTP/1.1.
```
Tried to log in <redacted> but could not find token  in database
```

**Comment by nvskamesh** [comment]
I had this same issue for almost a month now.  It would happen sporadically.  Restarting the Nextcloud server didn't make a difference.  It would not work on Safari, but work on Firefox and Safari private window.  I bumped into this thread via Google search and taking queues from here, I tried disabling "Cache Assets"option in my Nginx Proxy Manager.  I didn't even clear the safari cookies and website data and the login started working now.  I don't get this error anymore.

**Comment by kbftech** [comment]
Disabling "cache assets" in Nginx Proxy Manager did the trick here too. Thanks!

**Comment by cowpiebrownrice** [comment]
Same problem for me:
Nextcloud 30.0.8, docker installation.
PHP v.8.2.28

Nextcloud is behind a NGINX reverse proxy with **cache assets** disabled.

Problem was only with Chrome, already reset profile anche cache. With Edge no problem.

Every login it make an other token

![Image](https://github.com/user-attachments/assets/7a4b22c4-40f1-46ac-ba44-454f42d9d0b9)

![Image](https://github.com/user-attachments/assets/44df232a-8b31-487e-b113-82df54254d26)

Log:
```
{"reqId":"TYegTjlRnOohCc4qbD0A","level":1,"time":"2025-06-04T10:13:30+00:00","remoteAddr":"192.168.32.56","user":"--","app":"core","method":"GET","url":"/apps/theming/manifest?v=dd18bc46","message":"Tried to log in but could not verify token","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36","version":"30.0.8.1","data":{"app":"core","user":"utente_9058"}}
```

**Comment by Jaffex** [comment]
I'll comment on this issue since it's probably related.

I'm using nextcloud mostly for myself, but recently opened a public share, protected by a password, for a group of friends to upload files.

Some users reported problems when accessing the share, where it would not display files that were (from my perspective) clearly present.

When trying to find out what was happening, I encountered the same problem and tried to diagnose it.

The problem only seems to happen with mobile browsers, but only seems to appear after some time of accessing the share, probably after the granted token for a successful password entry expires.

After the token expired, the user will be asked for the public share's password. Providing the right password will grant the user access to the share, however in the logs, the following message is logged:

```
Info     core               Tried to log in but could not verify token                                                                                              2025-07-13T12:25:56+00:00
```

The user can then navigate around within the share, but certain events, like scrolling inside a large folder of images, will trigger the above log message quickly and repeatedly, until brute force will set an end to the user's experience:

```
Info     no app in context  IP address blocked because it reached the maximum failed attempts in the last 30 minutes [action: PublicPreviewController::getPreview,  2025-07-13T12:25:56+00:00
```

After this happens, the user will be unable to interact with files within the share anymore, but this is not even visible to them. To them it just seems that nextcloud is "stuck". Nothing seems to work anymore, even though the folder contents are being shown to them.

If they then try to access the public link from a private / incognito tab, the share will show 0 Files within the public share instead of notifying them of their blockage by brute force protection, completing the confusion.

Even if they try to access the share the next day, they will quickly get blocked by brute force again.

The only thing that helps is clearing the server data inside the mobile browser for my nextcloud instance, however instructing 20+ casual users to do so is not really a good solution.
